server/src/main/java/org/apache/druid/catalog/model/table/FormattedInputSourceDefn.java

```diff
@@ -95,7 +95,10 @@ protected List<ParameterDefn> addFormatParameters(
   )
   {
     final List<ParameterDefn> toAdd = new ArrayList<>();
-    final ParameterDefn formatProp = new Parameter(FORMAT_PARAMETER, ParameterType.VARCHAR, false);
+    // While the format parameter is required, we mark it as optional. Else
+    // if the source defines optional parameters, they will still be ignored
+    // as Calcite treats (optional, optional, required) as (required, required, required)
```

Comment:
What do you mean by this? Do you mean that if any of the parameters are required then all are?